frontend/src/components/associations/library/home/LoanableCard.tsx

```diff
@@ -1,48 +1,158 @@
-import React from "react";
+import React, { useContext } from "react";
 import { Loanable } from "../../../../models/associations/library";
 import Card from "react-bootstrap/Card";
-import { formatNewLines } from "../../../../utils/format";
+import { decidePlural, formatNewLines } from "../../../../utils/format";
 import { CardStatus } from "../../../utils/CardStatus";
 import { TablerColor } from "../../../../utils/colors";
 import Button from "react-bootstrap/Button";
 import dayjs from "dayjs";
+import { ToastContext } from "../../../utils/Toast";
+import { queryCache, useMutation } from "react-query";
+import { api } from "../../../../services/apiService";
+import { UserContext } from "../../../../services/authService";
+import OverlayTrigger from "react-bootstrap/OverlayTrigger";
+import Tooltip from "react-bootstrap/Tooltip";
+
+const LoanButton = ({
+  loanable,
+  loan,
+  cancel,
+}: {
+  loanable: Loanable;
+  loan: () => void;
+  cancel: () => void;
+}) => {
+  const CancelButton = ({ tooltipText }: { tooltipText: string }) => (
+    <OverlayTrigger
+      placement="bottom"
+      overlay={<Tooltip id={`${loanable.id}-tooltip`}>{tooltipText}</Tooltip>}
+    >
+      <Button className="btn-sm" variant="secondary" onClick={cancel}>
+        Annuler la demande
+      </Button>
+    </OverlayTrigger>
+  );
+
+  const LoanButton = ({ tooltipText }: { tooltipText: string }) => (
+    <OverlayTrigger
+      placement="bottom"
+      overlay={<Tooltip id={`${loanable.id}-tooltip`}>{tooltipText}</Tooltip>}
+    >
+      <Button className="btn-sm" variant="primary" onClick={loan}>
+        Emprunter
+      </Button>
+    </OverlayTrigger>
+  );
+
+  const UnavailableButton = () => (
```

Comment:
Maybe not useful to define the component to call it only once after that. It would be simpler to directly inject the code where you use the tag

Reply:
It's more explicit this way and does not break the symmetry with the other cases